[MRG] Updates on the scikit-learn sponsor page

cmarmo opened the pull request:
#### Reference Issues/PRs
Closes #14974 

#### What does this implement/fix? Explain your changes.
Updates the [funding](https://scikit-learn.org/stable/about.html#funding) section in the about us page.

#### TODO

Please, just comment about the content now: css will be fixed once the content well defined.
- [X] Separate active supporters from past supporters in two distinct sections
- [x] Decide whether we should keep or remove the logos on the home page:
  - If we keep them we should update the list: shall we expand the logos of all active supporters? If so, which order?
  - If we do not keep them we need to replace them by a link "scikit-learn development maintenance is financially supported by generous sponsors" or something similar
  - If we do not keep them we need to replace Google search with sphinx search
- [x] Improve the layout of the sponsor logos and text explanations in the funding section of about.html
- [x] Are there any institution / firm needs to be added to the list (past and present)?
- [x] Is the NumFocus Donate button still relevant? (will be updated in another PR if needed)

Thanks for your collaboration.

cmarmo committed:
Working on 'about' page.

cmarmo committed:
Merge branch 'master' into funders

cmarmo committed:
Version available for comments.

cmarmo committed:
Merge branch 'master' into funders

jnothman commented:
I like how this looks. Is it still WIP?

one area to improve: I think it's easy to currently misconstrue that the consortium is the primary/only funding source. So a preface might help clarify that several sources will be listed.

cmarmo committed:
Merge branch 'master' into funders

cmarmo committed:
Add a preface to the funder list.

cmarmo commented:
Thanks @jnothman for your answer.
I've added a small preface.
Yes, the PR is still WIP. I'm lacking some information.
I will add my questions directly in the code review.

cmarmo commented on doc/about.rst:
```diff
@@ -96,86 +96,176 @@ following paper:
-`The University of Sydney <https://sydney.edu.au/>`_ funds Joel Nothman since July 2017.
+   Andreas Müller received a grant to improve scikit-learn from the
+   `Alfred P. Sloan Foundation <https://sloan.org>`_ .
+   This grant supports the position of Nicolas Hug and Thomas J. Fan.
```
@amueller, @NicolasHug, @thomasjpfan could you please check?

cmarmo commented on doc/about.rst:
```diff
@@ -96,86 +96,176 @@ following paper:
+   :align: center
+   :target: https://sydney.edu.au/
+
+   `Anaconda, Inc <https://www.anaconda.com/>`_ funds Adrin Jalali since .
```
@adrinjalali could you please check? Thanks!

adrinjalali replied:
since 2019 (July)

cmarmo replied:
Thanks!!

cmarmo commented on doc/about.rst:
```diff
@@ -96,86 +96,176 @@ following paper:
+   <https://www.datascience-paris-saclay.fr/>`_
+   funded one year for a developer to work on the project full-time
+   (2014-2015), 50% of the time of Guillaume Lemaitre (2016-2017) and the
+   position of Joris van den Bossche (????-????).
```
Hello @jorisvandenbossche , could you please confirm the extent of your contract and the type (full or part time?)? Thanks!

jorisvandenbossche replied:
You can put the same as for Guillaume, but with years 2017-2018

cmarmo commented on doc/about.rst:
```diff
@@ -188,29 +278,43 @@ program.
+of the `PSF <https://www.python.org/psf/>`_ and `tinyclues
+<https://www.tinyclues.com/>`_.
 
 Donating to the project
```
See also the discussion about NumFocus here #6603
@GaelVaroquaux and Co. (I don't know exactly who I need to ping...) is that still relevant? Who is taking care of that?
Thanks!

cmarmo committed:
Removing logos from the homepage.

cmarmo commented:
Hello everybody: 2ca06131ab07bc58447a3b7ef377aca2ea11e286 I have removed the logos on the homepage and changed the link text in the community section.
It looks consistent to me , but please, feel free to comment.

I have now a concern about the Google search engine (#7061), because without any other logos in the homepage its presence ends up in a unjustified Google visibility with respect to other funders.
So please tell me... :) 
Either we re-add more logos (but then which one), either we switch to the sphinx search.
Thanks for your comments on that.

ogrisel commented on doc/about.rst:
```diff
@@ -96,86 +96,176 @@ following paper:
+.. figure:: images/bcg.png
+   :width: 100pt
+   :align: center
+   :target: https://www.bcg.com/beyond-consulting/bcg-gamma/default.aspx
```
I think we should find a way to pack the icons of the sponsors of the sckit-learn consortium on 3 lines (by sponsoring levels). I find that the current layout takes to much vertical real estate.

ogrisel commented on doc/about.rst:
```diff
@@ -96,86 +96,176 @@ following paper:
 
-.. image:: themes/scikit-learn/static/img/columbia.png
-   :width: 100pt
+**Silver Members**
```
We should find a way to scope those sections under the "scikit-learn consortium at Inria Foundation" group more explicitly.

Otherwise one gets the impression that the following supporters (Columbia U, Sloan, U. of Sydney...) are silver level members of the  Scikit-learn consortium at Inria Foundation.

ogrisel replied:
This is probably equivalent to what @jnothman expressed in his comment:

> one area to improve: I think it's easy to currently misconstrue that the consortium is the primary/only funding source. So a preface might help clarify that several sources will be listed.

ogrisel reviewed (commented):
Some comments on the logo layout:

cmarmo committed:
Consortium sponsors arranged in tables.

jnothman commented:
Yes, that's better, thanks!

Any reason this is still WIP?

thomasjpfan commented on doc/about.rst:
```diff
@@ -96,86 +96,180 @@ following paper:
+   :width: 100pt
+   :target: https://www.bcg.com/beyond-consulting/bcg-gamma/default.aspx
+
++-------+-------+
```
Is it possible to use:

```rst
.. table::
   :class: sk-sponser-table
```

To give the table a custom class, `sk-sponser-table`, so we can remove the blue background and the lines here:

<img width="339" alt="Screen Shot 2019-09-23 at 7 12 00 PM" src="https://user-images.githubusercontent.com/5402633/65469607-0bdace00-de36-11e9-8ce7-35e72a910ee2.png">

cmarmo replied:
done in ab7187829971db1cdafdafb0f0846c24cc44cb96

cmarmo committed:
Merge master, fix table style.

cmarmo commented:
> Any reason this is still WIP?

Changed to [MRG] . Someone will answer my questions in the review phase... right? :)

cmarmo committed:
Merge branch 'master' into funders

jnothman commented:
WIP is more or less equivalent to "do not merge". It means that merging in
the current state would provide something deficient.

ogrisel commented:
Summarizing some discussion we had yesterday IRL with @cmarmo @glemaitre and @jeremiedbb :

- [ ] The bold labels "Platinum Members", "Gold Members"... for the Inria Foundation sponsors are still too prominent . I personally think we should get rid of those labels to declutter the layout and keep the highlight on the text explanations.

- [ ] The layout "Logo" followed by centered text explanation that is used for most funding sources is revered for "Inria Foundation" because we need many "grouped" logos for one funding line in this case. Instead one could experiment with getting rid of the sphinx markup constraints and use inline HTML + new entries in the theme.css so as to have text on the left, logos on the right for all funding sources, irrespective of the number of logos on the right hand side.

- [ ] The Inria logo could be added to the list side-by-side with dataiku to keep the balance. The Inria support level is slightly different but let's keep things simple.

ogrisel commented:
I have refactored the description of the PR to turn it into a bullet point list. I think we should keep this PR as WIP as long as not all the boxes are ticked. Some of them just imply making a decision.

- For the NumFocus button I think we can leave it as it is. If it needs to be updated in anyway it can be done in a dedicated PR.

- For the home page, maybe we can try to expand the list of logos of all active supporters .

But then we have the problem of the order? Maybe oldest support to newest? Or shall we use a bit of javascript to shuffle the order on display?

cmarmo committed:
Merge branch 'master' into funders

GaelVaroquaux commented:
> Or shall we use a bit of javascript to shuffle the order on display?

I like the idea of shuffling.

cmarmo committed:
Tentative solution for logos display.

cmarmo committed:
Improve visibility of sponsor link on the homepage.

cmarmo commented:
Hi there, I've add the Inria Foundation logo because I did not like the version with two inria logos ... @ogrisel could you forgive me?
 
>     * For the NumFocus button I think we can leave it as it is. If it needs to be updated in anyway it can be done in a dedicated PR.

Ok, agree
 
>     * For the home page, maybe we can try to expand the list of logos of all active supporters .
> 
> 
> But then we have the problem of the order? Maybe oldest support to newest? Or shall we use a bit of javascript to shuffle the order on display?

Please... keep it simple... there is no equivalent among other python projects.
The home page is already so ... full ... simplicity is often the best solution...

jnothman commented:
I don't think the viewer recognises the shuffling. Just don't go there.

cmarmo committed:
Merge branch 'funders' of https://github.com/cmarmo/scikit-learn into funders

cmarmo committed:
Merge branch 'master' into funders

ogrisel commented:
Thanks @cmarmo, I really like the new layout with the logos on the right-hand side.

> I don't think the viewer recognizes the shuffling. Just don't go there.

Alright. I am also not a fan

So order by oldest (active) support to newest?
Or logo at all on the front page?

ogrisel commented:
+0 for to order by oldest (active) support to newest

jeremiedbb commented on doc/about.rst:
```diff
@@ -96,86 +96,339 @@ following paper:
+
+   The `Scikit-Learn Consortium at Inria Foundation
+   <https://scikit-learn.fondation-inria.fr/en/home/>`_ `Members 
+   <https://scikit-learn.fondation-inria.fr/en/home/#sponsors>`_ fund Olivier
```
```suggestion
   <https://scikit-learn.fondation-inria.fr/en/home/#sponsors>`_ funds Olivier
```

ogrisel replied:
May I suggest to rephrase as "The members of the Scikit-Learn Consortium at Inria Foundation fund...".

This would be easier to read and also visually highlights the fact that there are two links.

cmarmo replied:
done in aa269a0e886115a2da6dd77aee1993510bdfef22

ogrisel commented:
> Hi there, I've add the Inria Foundation logo because I did not like the version with two inria logos ...
> @ogrisel could you forgive me?

I will always forgive you, but there is nothing to forgive really :)

However puting the inria foundation logo on the left hand side breaks the consistency a bit. I would rather remove the Inria foundation logo on the left hand side and put the inria logo on the right hand side with the other foundation members.

@agramfort @GaelVaroquaux any opinion on this particular issue? Here is the current layout:


![Screenshot from 2019-09-26 10-39-15](https://user-images.githubusercontent.com/89061/65672837-fb9b3c80-e049-11e9-8e77-00223c619c50.png)

jeremiedbb commented on doc/about.rst:
```diff
@@ -96,86 +96,339 @@ following paper:
+   The `Scikit-Learn Consortium at Inria Foundation
+   <https://scikit-learn.fondation-inria.fr/en/home/>`_ `Members 
+   <https://scikit-learn.fondation-inria.fr/en/home/#sponsors>`_ fund Olivier
+   Grisel, Guillaume Lemaitre, Jérémie Du Boisberranger and Chiara Marmo.
```
```suggestion
   Grisel, Guillaume Lemaitre, Jérémie du Boisberranger and Chiara Marmo.
```
:)

cmarmo replied:
done in aa269a0e886115a2da6dd77aee1993510bdfef22

jeremiedbb reviewed (commented):
looking good !

cmarmo committed:
Implement some comments. More responsive design.

cmarmo committed:
Sorry Jeremie.

cmarmo commented:
> 
> However puting the inria foundation logo on the left hand side breaks the consistency a bit. I would rather remove the Inria foundation logo on the left hand side and put the inria logo on the right hand side with the other foundation members.

Ok, you win...
I have allowed the boxes to vertically align when the screen is small: I think this is more readable.
May I apply this flexbox scheme to the [testimonial](https://scikit-learn.org/stable/testimonials/testimonials.html) page?Its rendering on small screen is not optimal...

ogrisel commented:
> May I apply this flexbox scheme to the testimonial page?Its rendering on small screen is not optimal...

But maybe in another PR to keep this PR focused on the funding page.

ogrisel commented:
Maybe the logos of the consortium members would benefit from a bit of padding. They are a bit crammed at the moment. Otherwise this is looking good.

ogrisel commented:
> If we do not keep them we need to replace Google search with sphinx search

This is unrelated to the funding issue. I think google search works fine. Maybe sphinx search works better nowadays than it used to but this should better be discussed in another issue.

cmarmo commented:
> > May I apply this flexbox scheme to the testimonial page?Its rendering on small screen is not optimal...
> 
> I agree but maybe in another PR to keep this PR focused on the funding page.

Sure

cmarmo commented:
> > If we do not keep them we need to replace Google search with sphinx search
> 
> This is unrelated to the funding issue. I think google search works fine. Maybe sphinx search works better nowadays than it used to but this should better be discussed in another issue.

Sorry, but I do not agree with that: this is related if we do not add any logo to the page: Google will be the only firm cited on the homepage and this needs to be taken into account.

cmarmo committed:
Pad Consortium members.

ogrisel commented:
Let's keep the discussion for the search box to a dedicated PR. This is unrelated to how we acknowledge our funding sponsors.

cmarmo commented:
Hello everybody, my impression is that the homepage is a sensible matter ... I'd like to finalize the sponsor page in order to clarify things with respect to the Consortium engagements.
May I propose to revert the homepage at master and merge this PR? At least  those annoying security warnings are gone.. :)

ogrisel commented:
> May I propose to revert the homepage at master and merge this PR? At least those annoying security warnings are gone.. :)

I would rather not put back logos of past supporters while the newer supporters are not acknowledged. Can you please try to add all the logos of all the supporters in order from oldest to newest and see how that renders?

cmarmo committed:
Merge branch 'master' into funders

cmarmo committed:
Logos on the homepage.

cmarmo commented:
@ogrisel, I've added all the logos ... I'm not quite sure about the order ... let me know.

ogrisel reviewed (approved):
LGTM, thanks very much @cmarmo!

ogrisel commented:
Actually can you please add a link from the footer logos of the index.html page to the Funding section of the about.html page?

cmarmo committed:
Add link to funders.

cmarmo commented:
> Actually can you please add a link from the footer logos of the index.html page to the Funding section of the about.html page?

Done: but I'm really unhappy with the `text-decoration: underline` popping up when mouse-over the link... and can't find where this property is defined, and no time this evening to do that ... maybe @thomasjpfan has some ideas? Thanks!

ogrisel commented:
Maybe putting the `<a/>` inside the `<div/>` but around the `<img/>` tags?

thomasjpfan commented:
You can add a `class="sk-landing-funding-link"` to `a` and set:

```css
a.sk-landing-funding-link:hover {
  text-decoration: none;
}
```

cmarmo committed:
No underlined link to funders.

cmarmo commented:
> You can add a `class="sk-landing-funding-link"` to `a` and set:
> 
> ```css
> a.sk-landing-funding-link:hover {
>   text-decoration: none;
> }
> ```
Right! Thanks! Done.

jnothman commented:
I'm okay to get rid of the sponsors' logos on the homepage. Atm it it unclear that they are there because of sponsorship.

jnothman commented:
I like what's happening in about.html

cmarmo commented:
> I'm okay to get rid of the sponsors' logos on the homepage. Atm it it unclear that they are there because of sponsorship.

I agree with you... but I could not persuade @ogrisel ... :)

ogrisel commented:
>>  I'm okay to get rid of the sponsors' logos on the homepage. Atm it it unclear that they are there because of sponsorship.

> I agree with you... but I could not persuade @ogrisel ... :)

Alright, let's remove them from the homepage then. I did not want to change how we acknowledge support unilaterally.

cmarmo committed:
Merge branch 'master' into funders

cmarmo committed:
Remove logos from the homepage.

GaelVaroquaux commented:
>     I'm okay to get rid of the sponsors' logos on the homepage. Atm it it unclear that they are there because of sponsorship.

> I agree with you... but I could not persuade @ogrisel ... :)

Yesterday evening, I was thinking that it could get us in trouble with higher ups...

ogrisel commented:
> Yesterday evening, I was thinking that it could get us in trouble with higher ups...

What is more trouble: having logos on the homepage or not having logos?

GaelVaroquaux commented:
> What is more trouble: having logos on the homepage or not having logos?

Not having logos, I think. At least for us at Inria.

ogrisel commented:
Ok then we can re-add them with a short sentence such as:

"
[News box]                [Community box]                      [Testimonials box]


Scikit-learn development and maintenance is supported by generous sponsors:
[logo 1] [logo 2] ... [logo k] 
"

and we only keep the link to `about.html#funding` in the footer.

Previously this link was in the "Community" box of the homepage which was probably the source of confusion:

![Screenshot from 2019-10-08 15-49-28](https://user-images.githubusercontent.com/89061/66401317-466e6a00-e9e3-11e9-8e3f-4f17580aa8c1.png)

amueller commented:
looks good :)

cmarmo committed:
New suggestion

ogrisel commented:
Thanks @cmarmo for bearing with the decision process. Here is a screenshot of the current state:

![Screenshot from 2019-10-09 11-56-32](https://user-images.githubusercontent.com/89061/66471588-dff15680-ea8b-11e9-891f-e0dfa2037fee.png)

I am still +1 for merge :)

ogrisel commented:
Thank you very much @cmarmo! We can always refine in further PRs if needed.

GaelVaroquaux commented:
This is great! Thank you!
